### PR TITLE
fix error due to faulty dart_jsonwebtoken breaking change

### PR DESCRIPTION
**Pull request by ahmednfwela**
# Pull Request

## What does this PR do?
- rename `JWTError` to `JWTException` due to change in https://github.com/jonasroussel/dart_jsonwebtoken/issues/39 which was marked as a minor change, but it should've been a major one.
- add `example/pubspec.lock` to `.gitignore`

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,7 +10,7 @@
 .svn/
 .vscode/
 pubspec.lock
-!example/pubspec.lock
```
Comment (brunoocasali) [inline .gitignore:13]: I didn't know we could do it 👀.  I thought it was required by pub.dev

**Review by brunoocasali — APPROVED** [review]
LGTM!

**Comment by brunoocasali** [comment]
bors merge

File: .gitignore (hunk 1 above)
Comment (ahmednfwela) [reply .gitignore:13]: nope it's not required by pub.dev, and it's "best practice" to version control it in the final application, not in packages

**Pull request merged**